plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/manager/VmwareStorageManagerImpl.java

```diff
@@ -1464,4 +1448,55 @@ private String deleteDir(String dir) {
     private static String getVolumeRelativeDirInSecStroage(long volumeId) {
         return "volumes/" + volumeId;
     }
+
+    /**
+     * This method is to check if a given TaskInfo Object is valid( and has name and entity name assigned).It return true if TaskInfo Object is valid and false otherwise.
+     *
+     * @param TaskInfo
+     *        info
+     * @return boolean(true or false)
+     **/
+    private boolean isvalidTaskInfoObj(TaskInfo info){
+        return !(info == null || info.getEntityName() == null || info.getName() == null);
+    }
+    
+    /**
+     * This method waits for tasks running on vm to complete
+     *
+     * @param vmName        Name of the vm
+     * @param taskName      Name of the task
+     * @param context       Task context object
+     * 
+     * @return boolean(true or false)   True if it waited for tasks to finish and false when there are no tasks running on the Vm.
+     **/
+     
+    private boolean waitForRunningTaskOnVM(String vmName, String taskName, VmwareContext context) throws Exception {
```

Comment:
Throwing `Exception` is an anti-pattern and should be avoided.  Either throw a custom checked exception with contextual information and transform all checked exceptions to unchecked.